src/distro/distro.py

```diff
@@ -1055,13 +1056,17 @@ def _parse_os_release_content(lines: TextIO) -> Dict[str, str]:
             # stripped, etc.), so the tokens are now either:
             # * variable assignments: var=value
             # * commands or their arguments (not allowed in os-release)
+            # Ignore any tokens that are not variable assignments
             if "=" in token:
                 k, v = token.split("=", 1)
                 props[k.lower()] = v
-            else:
-                # Ignore any tokens that are not variable assignments
-                pass
 
+        if "version" in props:
+            # extract release codename (if any) from version attribute
+            match = re.search(r"\((\D+)\)|,\s*(\D+)", props["version"])
+            if match:
+                release_codename = match.group(1) or match.group(2)
+                props["codename"] = props["release_codename"] = release_codename
```

Comment:
@HorlogeSkynet maybe I missed it but it doesn't seem obvious: Do we even need this new field `release_codename` if…
- `props["codename"]` and therefore `_os_release_info["codename"]` and therfore `.codename()` all have the same right value already and
- it's not used anywhere but in `.linux_distribution()`.

Can it be resolved?

Reply:
@hartwork Actually the block you pointed out has been moved "up", but also an `elif` has disappeared : the idea of this patch is to "keep" `release_codename` (in OS-RELEASE(5) terms) available for further usages, even if `codename` is getting overwritten by `VERSION_CODENAME` (or its Ubuntu specific sibling).

I'd have preferred doing this parsing in `linux_distribution` method, but we wouldn't have the "raw" `VERSION` attribute value in such a scope.

**TL; DR** : the block has not _just_ been moved, the `release_codename` field "survives" #230 logic.
We can add an empty new line between `if` blocks if you think it would be appropriate.

Reply:
Definitely okay Sebastian ! 🎉

Reply:
Done ! I also took the liberty to precise a `linux_distribution` docstring bit, so a final proper review is required 😇